### PR TITLE
Improve condition

### PR DESCRIPTION
**Pull request by ngocnhan-tran1996**
This PR includes:
- Check `this.producer` before calling `lock()` to `createOrGetProducer()` and `close` method
- Modernize `switch` pattern
- Reduce `else` condition

### CODE REVIEW

**Comment by artembilan** [comment]
@ngocnhan-tran1996 , 
Thank you for contribution!
Looks like you are addressing the other PR : https://github.com/spring-projects/spring-amqp/pull/2862. Can yo mention that then and other contributor know, so we all don’t spend time looking into duplication?

**Review by artembilan — CHANGES_REQUESTED** [review]

**Review by artembilan — APPROVED** [review]
Pulling locally for final review and merge...

**Comment by artembilan** [comment]
Merged as https://github.com/spring-projects/spring-amqp/commit/33da2a6158a49ff62a37cbebc5ac1e8ac5e73839.

@ngocnhan-tran1996 ,

thank you for contribution; looking forward for more!

A couple remarks:

Why your fork is called `spring-amqp-fork`? Why it cannot be just `spring-amqp` in your repositories?
Please, make yourself familiar with this article to understand of the commit message importance: https://cbea.ms/git-commit/.
Plus, GitHub is able to turn the first commit of the PR to its description.
This way we can save some time for extra typing.

**Comment by artembilan** [comment]
This PR has also fixed https://github.com/spring-projects/spring-amqp/issues/2861.

**Comment by ngocnhan-tran1996** [comment]
Thanks @artembilan

I have a private repository name `spring-amqp` for some examples related `spring-amqp`. So I will rename it.

File: spring-rabbit-stream/src/main/java/org/springframework/rabbit/stream/producer/RabbitStreamTemplate.java [hunk 1]
```diff
@@ -107,29 +108,31 @@ public RabbitStreamTemplate(Environment environment, String streamName) {
...
-				else {
-					builder.superStream(this.streamName)
-							.routing(this.superStreamRouting);
-				}
-				this.producerCustomizer.accept(this.beanName, builder);
-				this.producer = builder.build();
-				if (!this.streamConverterSet) {
-					((DefaultStreamMessageConverter) this.streamConverter).setBuilderSupplier(
-							() -> this.producer.messageBuilder());
+		if (this.producer == null) {
```
[the remark this answers is not in the document]
Comment (artembilan) [reply spring-rabbit-stream/src/main/java/org/springframework/rabbit/stream/producer/RabbitStreamTemplate.java:111]: Done: https://github.com/spring-projects/spring-amqp/commit/f150bf531ae9cf0f8ae7e278889e7f9b084ffee6

**Pull request closed without merging**